node-api/backend/genesis.go

```diff
@@ -35,3 +35,31 @@ func (b Backend) GenesisValidatorsRoot(slot math.Slot) (common.Root, error) {
 	}
 	return st.GetGenesisValidatorsRoot()
 }
+
+// GenesisForkVersion returns the genesis fork version of the beacon chain.
+func (b Backend) GenesisForkVersion(genesisSlot math.Slot) (common.Version, error) {
+	st, _, err := b.stateFromSlot(genesisSlot)
+	if err != nil {
+		return common.Version{}, errors.Wrapf(err, "failed to get state from slot %d", genesisSlot)
+	}
+	fork, err := st.GetFork()
+	if err != nil {
+		return common.Version{}, errors.Wrapf(err, "failed to get fork from state")
+	}
+	return fork.CurrentVersion, nil
+}
+
+// GenesisTime returns the genesis time of the beacon chain.
+func (b Backend) GenesisTime(genesisSlot math.Slot) (math.U64, error) {
+	st, _, err := b.stateFromSlot(genesisSlot)
+	if err != nil {
+		return 0, errors.Wrapf(err, "failed to get state from slot %d", genesisSlot)
+	}
+	// Get the execution payload header from the beacon state,
+	// and return the timestamp as the genesis time.
+	execPayloadHeader, err := st.GetLatestExecutionPayloadHeader()
+	if err != nil {
+		return 0, errors.Wrapf(err, "failed to get execution payload header from state")
+	}
+	return execPayloadHeader.Timestamp, nil
+}
```

Comment:
This is probably the best we can do and it does not look too bad.
I think we should:
- compare what other API implementation do in Ethereum
- reach agreement about this and document it. cc @calbera

Reply:
Prysm stores the genesis time (and other misc info) in memory in various non-state enforced "helper" services

Reply:
2nd thing is we cannot query stateFromSlot with slot == 0 because comet's QueryContext function assumes height of 0 to be the head of the chain.

Reply:
So for Genesis Timestamp which is not stored in app-enforced state anywhere, we should load it into memory in the Backend upon genesis/initialization.

Reply:
I understood the 1st and 2nd pointer, Was trying to set genesis time in `init_chain.go`  in ProcessGenesisData , noticed that the ExecutionPayloadHeader does not have the timestamp. Might not  be the correct place.
Can you guide me from where to load it. DMed you the same. Thanks!

Reply:
Thanks for your help @calbera. As we discussed, now we are caching the genesis time from the genesis file and storing it in backend during initialisation.